tests/cpp/test_matmul_aten_evaluation.cpp

```diff
@@ -68,6 +68,44 @@ TEST_F(MatmulATenEvaluationTest, MmaOpAndCast) {
   EXPECT_TRUE(at::allclose(out[0], out_ref));
 }
 
+TEST_F(MatmulATenEvaluationTest, TransposeMmaOpAndCast) {
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  int64_t m = 32, n = 64, k = 128;
+  std::vector<int64_t> a_shape{m, k}, b_shape{n, k};
+
+  auto tv0 = makeConcreteTensor(a_shape, DataType::Half);
+  auto tv1 = makeConcreteTensor(b_shape, DataType::Half);
+  auto tv1_t = transpose(tv1, 0, 1);
+  auto tv0b = broadcast(tv0, {false, false, true}); // [M, K, 1]
+  auto tv1b = broadcast(tv1_t, {true, false, false}); // [1, K, N]
+  auto tv2 = fusedMultiplySum(tv0b, tv1b, {1});
+  auto tv3 = castOp(DataType::Half, tv2);
+
+  fusion->addInput(tv0);
+  fusion->addInput(tv1);
+  fusion->addOutput(tv3);
+
+  at::Tensor t0 = at::randn(a_shape, at::kHalf).cuda();
+  at::Tensor t1 = at::randn(b_shape, at::kHalf).cuda();
+  at::Tensor out_ref = at::matmul(t0, t1.transpose(0, 1));
+
+  FusionExecutorCache fec(std::move(fusion));
```

Comment:
Using the `FusionExecutorCache` here means you are also exercising the segmenter. That's fine if you don't suspect work will be needed on the segmenter, but in the future when exercising `linear`, I suggest creating a test that creates a `FusionExecutor` and marks aliasing for the outputs directly instead, so that you can avoid issues with segmentation and focus on getting ATen execution of linear working.